doc/sphinx/proof-engine/tactics.rst

```diff
@@ -105,6 +105,227 @@ bindings_list`` where ``bindings_list`` may be of two different forms:
   .. exn:: Not the right number of missing arguments.
      :undocumented:
 
+.. _intropatterns:
+
+Intro patterns
+~~~~~~~~~~~~~~
+
+Intro patterns let you specify the name to assign to variables and hypotheses
+introduced by tactics.  They also let you split an introduced hypothesis into
+multiple hypotheses or subgoals.  Common tactics that accept intro patterns
+include :tacn:`intros`, :tacn:`destruct` and :tacn:`apply`.
+
+.. productionlist::        coq
+   simple_intropattern     : `simple_intropattern_closed`
+                           : % `operconstr` ... % `operconstr`
```

Comment:
It's defined as: ```LIST0 ["%"; c = operconstr LEVEL "0" -> { c } ]"``` in simple_intropattern.

Reply:
BTW you should probably change this `operconstr` into `term` as it seems we rarely distinguish between the various term representations in the grammars in the manual.

Reply:
I'd like to understand that better first.  "term" is not even a non-terminal in the grammar.

Reply:
The internal name for "term" is "constr" in Coq for historical reason (when Coq was called "The Construction Engine" from the "Calculus of Constructions"). Developers have wondered whether it should be renamed to "term" but it is used too much for such a change to be worthwhile... However, in the manual we use "term" instead to be less confusing.

Reply:
The PR is still evolving, so maybe this is not yet the right time to comment. A few remarks though.

Yes, `operconstr` is the internal parser name of the recursive grammar of terms. This grammar has two useful entry points: `constr` which is `operconstr` at level 8 (to be used in positions where applications needs to be parenthesized to avoid collisions with the rest of the grammar), and `lconstr` which is `operconstr` at level 200 (to be used in a context where the expression is already surrounded by delimiters).

We could in principle be explicit in the reference manual on the different uses of level, but it has been considered simple to call `term` any of `constr`, `lconstr` and `operconstr`. Some improvements could be done though.